lib/utils/query-selection-set.ts

```diff
@@ -0,0 +1,14 @@
+export default function getQuerySelectionSet(query: Record<string, any>): Set<string> {
+  if (!query.select) {
+    return new Set()
+  }
+
+  // The selection of fields for the query is limited
+  // Get the different parts that are listed for selection
+  const allSelects = Array.isArray(query.select)
+    ? query.select
+    : query.select.split(',').map((q) => q.trim())
```

Comment:
nit: maybe `q` here could be `parameter` or even `param`?

Reply:
actually not sure if that's the best name, I'm just not a fan of single character variables. Just in case in future the function gets bigger and context can get lost